packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -47,25 +48,256 @@ def _create_file(
+            print(f"{index}. {display_name}")
+
+        selection = click.prompt(
+            "\nSelect a measurement to generate a client (x to exit)",
+            type=str,
+        )
+        if selection.lower() == "x":
+            break
+        service_class = _get_selected_measurement_service_class(
+            int(selection), measurement_service_class
+        )
+
+        base_service_class = _extract_base_service_class(service_class)
+        default_module_name = _create_module_name(base_service_class)
+        module_name = click.prompt(
+            "Enter a name for the Python client module (or) press enter to choose the default name",
+            type=str,
+            default=default_module_name,
+        )
+        _validate_identifier(module_name, "module")
+        default_class_name = _create_class_name(base_service_class)
+        class_name = click.prompt(
+            "Enter a name for the Python client class (or) press enter to choose the default name",
+            type=str,
+            default=default_class_name,
+        )
+        _validate_identifier(class_name, "class")
+
+        _create_client(
+            channel_pool=channel_pool,
+            discovery_client=discovery_client,
+            measurement_service_class=service_class,
+            module_name=module_name,
+            class_name=class_name,
+            directory_out=directory_out_path,
+        )
+
+
+def _create_clients(
+    measurement_service_class: List[str],
+    module_name: Optional[str],
+    class_name: Optional[str],
+    directory_out: Optional[str],
+) -> None:
+    channel_pool = GrpcChannelPool()
+    discovery_client = DiscoveryClient(grpc_channel_pool=channel_pool)
+    directory_out_path = _resolve_output_directory(directory_out)
+
+    is_multiple_measurement_client_creation = len(measurement_service_class) > 1
+    for service_class in measurement_service_class:
+        base_service_class = _extract_base_service_class(service_class)
+        if is_multiple_measurement_client_creation or module_name is None:
+            module_name = _create_module_name(base_service_class)
+        if is_multiple_measurement_client_creation or class_name is None:
+            class_name = _create_class_name(base_service_class)
+        _validate_identifier(module_name, "module")
+        _validate_identifier(class_name, "class")
+
+        _create_client(
+            channel_pool=channel_pool,
+            discovery_client=discovery_client,
+            measurement_service_class=service_class,
+            module_name=module_name,
+            class_name=class_name,
+            directory_out=directory_out_path,
+        )
+
+
 @click.command()
-@click.argument("measurement_service_class", nargs=-1)
-@click.option(
+@optgroup.group(
+    "all-modes",
+    cls=RequiredMutuallyExclusiveOptionGroup,
+    help="The different modes to create Python measurement client.",
+)
+@optgroup.option(
+    "-s",
+    "--measurement_service_class",
```

Comment:
Long options should use kebab-case, not snake_case. Click will automatically convert to snake_case variable names.
```suggestion
    "--measurement-service-class",
```